x-pack/libbeat/management/managerV2.go

```diff
@@ -251,38 +330,32 @@ func (cm *BeatV2Manager) unitListen() {
 				cm.logger.Debug("Received sighup, stopping")
 			}
 			cm.isRunning = false
-			unit, mainExists := cm.getMainUnit()
-			if mainExists {
-				_ = unit.UpdateState(client.UnitStateStopping, "stopping beat", nil)
-			}
-			cm.client.Stop()
+			cm.UpdateStatus(lbmanagement.Stopping, "Stopping")
 			return
 		case change := <-cm.client.UnitChanges():
 			switch change.Type {
 			// Within the context of how we send config to beats, I'm not sure there is a difference between
 			// A unit add and a unit change, since either way we can't do much more than call the reloader
 			case client.UnitChangedAdded:
-				// At this point we also get a log level, however I'm not sure the beats core logger provides a
-				// clean way to "just" change the log level, without resetting the whole log config
-				state, _, _ := change.Unit.Expected()
-				cm.logger.Debugf("Got unit added: %s, type: %s expected state: %s", change.Unit.ID(), change.Unit.Type(), state.String())
-				go cm.handleUnitReload(change.Unit)
-
+				cm.addUnit(change.Unit)
+				t.Reset(changeDebounce)
```

Comment:
Can the timer be running here? If it isn't stopped the timer documentation says this isn't the correct way to handle this: https://pkg.go.dev/time#Timer.Reset

> For a Timer created with NewTimer, Reset should be invoked only on stopped or expired timers with drained channels.
> 
> If a program has already received a value from t.C, the timer is known to have expired and the channel drained, so t.Reset can be used directly. If a program has not yet received a value from t.C, however, the timer must be stopped and—if Stop reports that the timer expired before being stopped—the channel explicitly drained:
> 
> if !t.Stop() {
> 	<-t.C
> }
> t.Reset(d)
> This should not be done concurrent to other receives from the Timer's channel.

Reply:
I looked into this and it seems the way I have it in the code is correct. The reason it is correct is because the `<-t.C` is handled in the select in the same goroutine. It that was not the case it would be a problem.

By using the code as you commented above doesn't work because the `<-t.C` would then occur in multiple places and 2 places cannot read from that channel or it will cause the one that doesn't get the message from the channel to block.

Many instances of golang code use the `t.Reset` like I do in the select and seems to work as expected in the code.